apt-hook/json-hook-src/json-hook.go

```diff
@@ -0,0 +1,253 @@
+	Id           int    `json:"id"`
+	Version      string `json:"version"`
+	Architecture string `json:"architecture"`
+	Pin          int    `json:"pin"`
+	Origins      []struct {
+		Archive  string `json:"archive"`
+		Codename string `json:"codename"`
+		Version  string `json:"version"`
+		Origin   string `json:"origin"`
+		Label    string `json:"label"`
+		Site     string `json:"site"`
+	} `json:"origins"`
+}
+type jsonRPC struct {
+	JsonRPC string `json:"jsonrpc"`
+	Method  string `json:"method"`
+	Params  struct {
+		Command         string   `json:"command"`
+		UnknownPackages []string `json:"unknown-packages"`
+		Packages        []struct {
+			Id           int    `json:"id"`
+			Name         string `json:"name"`
+			Architecture string `json:"architecture"`
+			Mode         string `json:"mode"`
+			Automatic    bool   `json:"automatic"`
+			Versions     struct {
+				Candidate jsonRPCPackageVersion `json:"candidate"`
+				Install   jsonRPCPackageVersion `json:"install"`
+				Current   jsonRPCPackageVersion `json:"current"`
+			} `json:"versions"`
+		} `json:"packages"`
+	} `json:"params"`
+}
+
+func createUpdateMessage(standardSecurityCount int, esmInfraCount int, esmAppsCount int) string {
+	displayStandard := true
+	displayEsmInfra := true
+	displayEsmApps := true
+	if standardSecurityCount == 0 {
+		displayStandard = false
+	}
+	if esmInfraCount == 0 {
+		displayEsmInfra = false
+	}
+	if esmAppsCount == 0 {
+		displayEsmApps = false
+	}
+
+	if !displayStandard && !displayEsmInfra && !displayEsmApps {
+		return ""
+	}
+
+	esmInfraFirst := false
+	esmAppsFirst := false
+	if !displayStandard && displayEsmInfra {
+		esmInfraFirst = true
+	} else if !displayStandard && !displayEsmInfra && displayEsmApps {
+		esmAppsFirst = true
+	}
+
+	standardUpdates := ""
+	esmInfraUpdates := ""
+	esmAppsUpdates := ""
+	if displayStandard {
+		standardUpdates = fmt.Sprintf("%d standard security ", standardSecurityCount)
+		if standardSecurityCount > 1 {
+			standardUpdates += "updates"
+		} else {
+			standardUpdates += "update"
+		}
+		if displayEsmInfra && displayEsmApps {
+			standardUpdates += ","
+		}
+		if displayEsmInfra || displayEsmApps {
+			standardUpdates += " "
+		}
+		if (displayEsmInfra && !displayEsmApps) || (!displayEsmInfra && displayEsmApps) {
+			standardUpdates += "and "
+		}
+	}
+	if displayEsmInfra {
+		esmInfraUpdates = fmt.Sprintf("%d esm-infra ", esmInfraCount)
+		if esmInfraFirst {
+			esmInfraUpdates += "security "
+		}
+		if esmInfraCount > 1 {
+			esmInfraUpdates += "updates"
+		} else {
+			esmInfraUpdates += "update"
+		}
+		if displayEsmApps {
+			esmInfraUpdates += " and "
+		}
+	}
+	if displayEsmApps {
+		esmAppsUpdates = fmt.Sprintf("%d esm-apps ", esmAppsCount)
+		if esmAppsFirst {
+			esmAppsUpdates += "security "
+		}
+		if esmAppsCount > 1 {
+			esmAppsUpdates += "updates"
+		} else {
+			esmAppsUpdates += "update"
+		}
+	}
+
+	return standardUpdates + esmInfraUpdates + esmAppsUpdates
+}
```

Comment:
It feels like this section if fairly repetitive.  It's good for this iteration and PR, but I'm wondering if we can iterate to do something like concatonate_strings([<updates_string>, <infra_str>, <apps_str]) and just have the comma spacing additions based on number of items in a list

Reply:
This section is definitely repetitive, but I tried to balance that with the readability. The nice thing about this construction is that you can read it top to bottom and see what the message will be, rather than adding things in the middle later.

I think its worth revisiting this, but I'd rather take it as a follow up item if thats okay with you

Reply:
+1 For sure. This is good for today! Thanks again